camayoc/tests/qpc/cli/test_scans.py

```diff
@@ -428,5 +428,10 @@ def test_clear_all(isolated_filesystem, qpc_server_config, data_provider):
 
     # Remove scan
     result = scan_clear({"all": None})
-    match = re.match(r"All scans were removed.", result)
-    assert match is not None
+    # Note! When run in isolation, this test may delete only one scan.
+    # However, when part of the full test suite, scans from previous tests may
+    # be lingering, and calling "clear --all" here may delete more than just the
+    # one scan added in the lines above. For example, I saw this output in our
+    # test pipeline: Successfully deleted 18 scans.
```

Comment:
Would it make sense to run this test at the beginning of suite execution, like we do for credentials https://github.com/quipucords/camayoc/blob/main/camayoc/tests/conftest.py#L10 ?

AFAIR we call delete all on credentials first, because there might be sources depending on these credentials. I think nothing depends on scans, so they can be removed anytime?

Reply:
I had a similar thought too, but I don't have strong feelings about this either way. Yes, scans are like leaf nodes in a tree of dependencies here. You can (or _should_ be able to) clear all scans without any being skipped due to dependencies from other objects. The backend server actually deletes some related objects (connection results, tasks, jobs) to ensure that the scans can safely be deleted, unlike the related bulk delete operations for sources and credentials. So, it seems safe enough to me to run this test anywhere, as long as tests are run serially and never in parallel, which I believe will always be true to to the nature of this test suite.

My current thought was to ignore this concern until later if/when it becomes more of a nuisance, but I would defer to your expertise if you think we should rearrange the order of operations now.

Reply:
Nah, I think if pipeline works without messing up with tests order, then let's keep the current order.

Sometimes I think about parallelization (that's probably one place were we could gain the biggest performance wins), but we're not going to run everything in parallel without some serious engineering effort. I don't see that happening in any foreseeable future. So assumption that tests are going to run serially is going to hold true for a while.